controllers/provisioners/eks/helpers.go

```diff
@@ -1173,3 +1173,40 @@ func (ctx *EksInstanceGroupContext) GetDesiredMixedInstancesPolicy(name string)
 
 	return policy
 }
+
+func filterSupportedArch(architectures []*string) string {
+	supportedArchitectures := []string{"x86_64", "arm64"}
+
+	for _, a := range architectures {
+		for _, supportedArch := range supportedArchitectures {
+			result := *a == supportedArch
+			if result {
+				return supportedArch
+			}
+		}
+	}
+	return ""
+}
+
+func (ctx *EksInstanceGroupContext) GetEksLatestAmi() (string, error) {
+	var (
+		instanceGroup = ctx.GetInstanceGroup()
+		state         = ctx.GetDiscoveredState()
+		configuration = instanceGroup.GetEKSConfiguration()
+	)
+	clusterVersion := state.GetClusterVersion()
+	annotations := instanceGroup.GetAnnotations()
+	var OSFamily string
+	if annotations[OsFamilyAnnotation] != "" {
```

Comment:
could probably move this logic into a helper function - I believe it exists elsewhere as well

Reply:
See `HasAnnotation` helper